### PR TITLE
ci: Add cluster profiles to `run-script` workflow

### PR DESCRIPTION
**Pull request by raunakab**
# Overview

This PR introduces cluster-profile to the run-script workflow. The profiles are:
- debug_xs-x86
- medium-x86
- large-x86

### CODE REVIEW

**Comment by raunakab** [comment]
Example of a successful workflow run (using the `debug_xs-x86`) configuration:
https://github.com/Eventual-Inc/Daft/actions/runs/12324000427

File: ci/assets/large-x86.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      KeyName: ci-github-actions-ray-cluster-key
+      InstanceType: i3.4xlarge
+      ImageId: ami-04dd23e62ed049936
+      IamInstanceProfile:
+        Name: ray-autoscaler-v1
+
+  ray.worker.default:
+    min_workers: 8
+    max_workers: 8
+    resources: {CPU: 100}
```
Comment (colin-ho) [inline ci/assets/large-x86.yaml:28]: Why limit to 100?

**Comment by raunakab** [comment]
@colin-ho Example of a successful review using the `large-x86` cluster profile:
https://github.com/Eventual-Inc/Daft/actions/runs/12324279856/job/34401526669

File: ci/assets/large-x86.yaml (hunk 1 above)
Comment (raunakab) [reply ci/assets/large-x86.yaml:28]: Is there a way to specify minimum CPUs? I couldn't find that in the docs [here](https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#cluster-configuration-resources-type).

**Review by jaychia — CHANGES_REQUESTED** [review]
#3568 was not approved, this PR will have to be rebased onto the new approach

**Pull request closed without merging**